src/mainwindow.cpp

```diff
@@ -42,8 +40,9 @@ void MainWindow::on_actionCompress_triggered()
 
 void MainWindow::Extract()
 {
-    extractWindow->show();
-    this->hide();
+    ExtractWindow extractWindow(new ExtractWindow(this));
```

Comment:
I'm not sure this is true for initialization, I think this will caused memory leak.

Reply:
Yes that is a mistake.